dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -982,6 +982,91 @@ TRASH_TrashFile(LPCWSTR wszPath)
     return DeleteFileToRecycleBin(wszPath);
 }
 
+static void TRASH_PlayEmptyRecycleBinSound()
+{
+    LPCWSTR lpRegSubKey = L"AppEvents\\Schemes\\Apps\\Explorer\\EmptyRecycleBin";
+    HKEY hRegKey, hRegSnd;
+    LPWSTR lpRegVal, lpSndPath;
+    DWORD dwRegValType, dwRegValSize, dwSndPathLen;
+    LONG lRegStatus;
+
+    /* Open registry key */
+    lRegStatus = RegOpenKeyExW(HKEY_CURRENT_USER, lpRegSubKey, 0, KEY_QUERY_VALUE, &hRegKey);
```

Comment:
Isn't it depends on ATL? Why create more dependency?

Reply:
Yes, atl's CRegKey depends on ATL.
But how is this adding a dependency?
Most of our shell code already uses ATL because having to write the same 100 lines of code for every object is not a job for a human, so we use a library for that.

Reply:
OK, I get it, but I don't think CRegKey will reduce the lines of my code so much compared to WinAPI functions (WinAPI just requires more function arguments) and I have more control using WinAPI.